translation-service-eTranslation/src/main/java/eu/europeana/api/translation/service/etranslation/ETranslationTranslationService.java

```diff
@@ -0,0 +1,211 @@
+import eu.europeana.api.translation.service.exception.TranslationException;
+
+public class ETranslationTranslationService extends AbstractTranslationService {
+  
+  private static final Logger LOGGER = LogManager.getLogger(ETranslationTranslationService.class);
+
+  private String serviceId;
+  private final String baseUrl;
+  private final String domain;
+  //this is the base url of the translation api (without the request handler (or the controller) endpoint)
+  private final String callbackUrl;
+  private final String credentialUsername;
+  private final String credentialPwd;
+  private final int maxWaitMillisec;
+  private final RedisMessageListenerContainer redisMessageListenerContainer;
+  
+  public ETranslationTranslationService(String baseUrl, String domain, String callbackUrl, int maxWaitMillisec, 
+      String username, String password, RedisMessageListenerContainer redisMessageListenerContainer) {
+    this.baseUrl = baseUrl;
+    this.domain = domain;
+    this.callbackUrl=callbackUrl;
+    this.maxWaitMillisec=maxWaitMillisec;
+    this.credentialUsername=username;
+    this.credentialPwd=password;
+    this.redisMessageListenerContainer=redisMessageListenerContainer;
+  }
+
+  @Override
+  public void translate(List<TranslationObj> translationObjs) throws TranslationException {
+    //base64 encoded string, which will not be translated from eTransaltion, used as a markup delimiter
+    String markupDelimit=" ZXd3ZHdld2U= ";
+    String eTranslJointStr = generateJointStringForTranslation(translationObjs, markupDelimit);
+    
+    /* create an eTransl request with an external reference and send it. The same external reference is received
+     * in the eTransl callback. That reference is used for the name of the channel for the redis message subscriber 
+     * listener created below, which will be notified from the redis publisher after the eTransl callback comes.
+     * The publisher will publish to the same channel using the external reference from the eTransl callback.
+     */
+    if(StringUtils.isNotBlank(eTranslJointStr)) {
+      //create external reference for eTransl service
+      byte[] eTranslExtRefBase64 = null;
+      try {
+        eTranslExtRefBase64 = Base64.encodeBase64(eTranslJointStr.getBytes(StandardCharsets.UTF_8.name()));
+      } catch (UnsupportedEncodingException e) {
+        throw new TranslationException("Exception during the eTranslation base64 encoding for the external reference.", 0, e);
+      }
+      String eTranslExtRef = new String(eTranslExtRefBase64, StandardCharsets.UTF_8);
+
+      //create and send the eTransl request
+      //baseUrl is empty for the integration tests, where the eTranslation service will not be called
+      if(!StringUtils.isBlank(baseUrl)) {
+        try {
+          String body = createTranslationBody(eTranslJointStr,translationObjs.get(0).getSourceLang(),translationObjs.get(0).getTargetLang(),eTranslExtRef);
```

Comment:
I assume that src language is mandatory for eTranslation, we need to prevent sending invalid requests to eTranslation